src/gwt/src/org/rstudio/studio/client/workbench/views/connections/ConnectionsPresenter.java

```diff
@@ -248,123 +247,61 @@ public void onActiveConnectionsChanged(ActiveConnectionsChangedEvent event)
    {
       updateActiveConnections(event.getActiveConnections());
    }
+
+   private void terminateShinyApp()
```

Comment:
Interrupting R is asynchronous and might not succeed -- In the context used here (interrupt, then perform an action) I think `ApplicationInterrupt.interruptR` is what you need.